tests/nimony/sysbasics/tdefaultparams.nim

```diff
@@ -0,0 +1,37 @@
+proc foo1(x: int, y: float = 1.2) =
+  let m = x
+
+proc foo2(x = 3, y = 7) =
+  let s = x
+  let j = y
+
+proc foo3(x: int, y: float, z: int = 23) =
+  discard
+
+proc foo4(x: int, y: float, z: int = 23, a: float = 1.2) =
+  discard
+
+proc foo5[T](x: T, y: int = 7) =
+  discard
+
+proc foo6[T](x: T = 3, y: int = 7) =
+  discard
+
+proc foo[T](x: T, y: T = 7) =
```

Comment:
But there is no reason this should compile as we strive to type check generics and "some T" is not necessarily compatible with the number 7.